File: packages/parser-ts/src/__fixtures__/dog.ts
import { Animal, Feedable } from "./animal";

export class Dog extends Animal implements Feedable {
  private breed: string;

  constructor(name: string, breed: string) {
    super(name);
    this.breed = breed;
  }

  speak(): string {
    return "Woof";
  }

  feed(amount: number): void {
    // no-op for fixture purposes
  }
}

File: packages/parser-java/src/__fixtures__/src/main/java/com/example/animals/Dog.java
package com.example.animals;

public class Dog extends Animal implements Feedable {
    private String breed;

    public Dog(String name, String breed) {
        super(name);
        this.breed = breed;
    }

    @Override
    public String speak() {
        return "Woof";
    }

    @Override
    public void feed(int amount) {
    }

    public static Dog create(String name) {
        return new Dog(name, "unknown");
    }
}
